src/ProjectService.Business/Commands/Project/EditProjectCommand.cs

```diff
@@ -1,142 +1,91 @@
-            _userRepository = userRepository;
-            _rcGetDepartments = rcGetDepartments;
-            _httpContextAccessor = httpContextAccessor;
-            _logger = logger;
-        }
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.Add("Not enough rights.");
 
-        public OperationResultResponse<bool> Execute(Guid projectId, JsonPatchDocument<EditProjectRequest> request)
-        {
-            _validator.ValidateAndThrowCustom(request);
+        return response;
+      }
 
-            DbProject dbProject = _projectRepository.Get(new GetProjectFilter { ProjectId = projectId });
+      if (!_validator.ValidateCustom(request, out List<string> errors))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
 
-            OperationResultResponse<bool> response = new();
-            Guid userId = _httpContextAccessor.HttpContext.GetUserId();
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors = errors;
 
-            if (!_accessValidator.IsAdmin() &&
-                !_userRepository.AreUserProjectExist(userId, projectId, true) &&
-                GetDepartment(dbProject.DepartmentId, response.Errors)?.DirectorUserId != userId)
-            {
-                throw new ForbiddenException("Not enough rights.");
-            }
+        return response;
+      }
 
-            foreach (Operation item in request.Operations)
-            {
-                if (item.path == $"/{nameof(EditProjectRequest.Name)}" &&
-                    _projectRepository.IsProjectNameExist(item.value.ToString()))
-                {
-                    response.Status = OperationResultStatusType.Failed;
-                    response.Errors.Add($"Project with name '{item.value}' already exist");
-                    return response;
-                }
+      response.Body = _projectRepository.Edit(projectId, _mapper.Map(request));
+      if (!response.Body)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
 
-                if (item.path == $"/{nameof(EditProjectRequest.DepartmentId)}")
-                {
-                    // TODO rework to department existence
-                    var departmentData = GetDepartment(Guid.Parse(item.value.ToString()), response.Errors);
+        response.Status = OperationResultStatusType.Failed;
+        response.Errors.Add("Project can not be edit.");
 
-                    if (!response.Errors.Any() && departmentData == null)
-                    {
-                        throw new BadRequestException("Project department not found.");
-                    }
-                    else if (response.Errors.Any())
-                    {
-                        response.Status = OperationResultStatusType.Failed;
-                        return response;
-                    }
-                };
-            }
+        return response;
```

Comment:
Delete this return and move line 86 to line 76

Reply:
done